platform/linuxbsd/joypad_linux.cpp

```diff
@@ -183,11 +184,20 @@ void JoypadLinux::monitor_joypads() {
 		{
 			MutexLock lock(joy_mutex);
 
-			for (int i = 0; i < 32; i++) {
```

Comment:
If I understood correctly what you are suggesting, it won't work.  E.g. in my laptop's case

```
[ronchaine@mellori (130) ~]$ evtest
No device specified, trying to scan all of /dev/input/event*
Not running as root, no devices may be available.
Available devices:
/dev/input/event0:	Lid Switch
/dev/input/event1:	Power Button
/dev/input/event2:	AT Translated Set 2 keyboard
/dev/input/event3:	Intel HID events
/dev/input/event4:	Intel HID 5 button array
/dev/input/event5:	Intel Virtual Button driver
/dev/input/event6:	Integrated IR Camera: Integrate
/dev/input/event7:	Integrated Camera: Integrated C
/dev/input/event8:	PC Speaker
/dev/input/event9:	ThinkPad Extra Buttons
/dev/input/event10:	Video Bus
/dev/input/event11:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3
/dev/input/event12:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3 System Control
/dev/input/event13:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3
/dev/input/event14:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3 Consumer Control
/dev/input/event15:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3 Consumer Control
/dev/input/event16:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3 Wireless Radio Control
/dev/input/event17:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3
/dev/input/event18:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3
/dev/input/event19:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3
/dev/input/event20:	HDA Digital PCBeep
/dev/input/event21:	HDA Intel PCH Headphone
/dev/input/event22:	HDA Intel PCH HDMI/DP,pcm=7
/dev/input/event23:	HDA Intel PCH Mic
/dev/input/event24:	HDA Intel PCH HDMI/DP,pcm=3
/dev/input/event25:	HDA Intel PCH HDMI/DP,pcm=8
/dev/input/event26:	HDA Intel PCH HDMI/DP,pcm=9
/dev/input/event27:	HDA Intel PCH HDMI/DP,pcm=10
/dev/input/event28:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3 Mouse
/dev/input/event29:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3 Mouse
/dev/input/event30:	Chicony ThinkPad X1 Tablet Thin Keyboard Gen 3 Touchpad
/dev/input/event31:	Wacom HID 511A Pen
/dev/input/event256:	Wacom HID 511A Finger
/dev/input/event257:	Sony Computer Entertainment Wireless Controller Touchpad
/dev/input/event258:	Sony Computer Entertainment Wireless Controller Motion Sensors
/dev/input/event259:	Sony Computer Entertainment Wireless Controller
```
If I got your suggestion right, it would break at `event32` and the controllers would still be missed.

Reply:
You're right. Then let me now stop suggesting that. 😀